### ISSUE TITLE
Devices with no capture formats cause panic

### ISSUE THREAD

**Issue by laptou** [raymanfx/libv4l-rs#8]
- Trying to enumerate the capture formats on devices that have none causes a panic.

Minimal reproducible example:

```rust
extern crate clap;
extern crate v4l;

use clap::{App, Arg};
use v4l::prelude::*;

fn main() {
    let matches = App::new("v4l device")
        .version("0.2")
        .author("Christopher N. Hesse <raymanfx@gmail.com>")
        .about("Video4Linux device example")
        .arg(
            Arg::with_name("device")
                .short("d")
                .long("device")
                .value_name("INDEX or PATH")
                .help("Device node path or index (default: 0)")
                .takes_value(true),
        )
        .get_matches();

    // Determine which device to use
    let mut path: String = matches
        .value_of("device")
        .unwrap_or("/dev/video0")
        .to_string();

    if path.parse::<u64>().is_ok() {
        path = format!("/dev/video{}", path);
    }

    println!("Using device: {}\n", path);

    let device = CaptureDevice::with_path(path).unwrap();

    println!("Formats:");
    
    for fmt in device.enum_formats().unwrap() {
        println!("\t{:#?}", fmt.description);
        println!("\t\tflags: {:#}", fmt.flags);
        println!("\t\tformat: {:#}", fmt.fourcc);
    }
}
```

Output:
```
v4l-rs-bug-repro is 📦 v0.1.0 via 🦀 v1.45.2 
❯ RUST_BACKTRACE=1 cargo run -- --device=/dev/video2
   Compiling v4l-rs-bug-repro v0.1.0 (/home/ibiyemi/projects/scratch/v4l-rs-bug-repro)
    Finished dev [unoptimized + debuginfo] target(s) in 0.56s
     Running `target/debug/v4l-rs-bug-repro --device=/dev/video2`
Using device: /dev/video2

Formats:
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 22, kind: InvalidInput, message: "Invalid argument" }', src/main.rs:38:16
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1076
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1537
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:218
  10: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:486
  11: rust_begin_unwind
             at src/libstd/panicking.rs:388
  12: core::panicking::panic_fmt
             at src/libcore/panicking.rs:101
  13: core::option::expect_none_failed
             at src/libcore/option.rs:1272
  14: core::result::Result<T,E>::unwrap
             at /home/ibiyemi/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/result.rs:1005
  15: v4l_rs_bug_repro::main
             at src/main.rs:38
  16: std::rt::lang_start::{{closure}}
             at /home/ibiyemi/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/rt.rs:67
  17: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:52
  18: std::panicking::try::do_call
             at src/libstd/panicking.rs:297
  19: std::panicking::try
             at src/libstd/panicking.rs:274
  20: std::panic::catch_unwind
             at src/libstd/panic.rs:394
  21: std::rt::lang_start_internal
             at src/libstd/rt.rs:51
  22: std::rt::lang_start
             at /home/ibiyemi/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libstd/rt.rs:67
  23: main
  24: __libc_start_main
  25: _start
```

Where `/dev/video2` is a `v4l2loopback` device with no capture formats.

**Comment by raymanfx** [comment]
I'll take a look at this tomorrow.

**Comment by raymanfx** [comment]
Hmm, thinking about this some more, I believe the Linux v4l2 API design is a bit lacking here.
From [1]: EINVAL - The struct v4l2_fmtdesc type is not supported or the index is out of bounds.
It might be an option to keep the current behavior, because we can't know whether the type is not supported or whether there are no formats.

[1] https://www.kernel.org/doc/html/v4.14/media/uapi/v4l/vidioc-enum-fmt.html

**Comment by raymanfx** [comment]
Fixed in https://github.com/raymanfx/libv4l-rs/commit/176f88ddd586d89c9906cc1ad70adcf1fcc3d992 (released as part of 0.9.2).
